src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs

```diff
@@ -216,31 +212,28 @@ void M(int [|p|], bool flag, bool flag2)
         [Fact]
         public async Task OptionalParameter_Unused()
         {
-            await TestDiagnosticsAsync(
-                """
+            await VerifyAnalyzerAsync("""
                 class C
                 {
-                    void M(int [|p|] = 0)
+                    void M(int {|IDE0060:p|} = 0)
                     {
                     }
                 }
-                """,
-    Diagnostic(IDEDiagnosticIds.UnusedParameterDiagnosticId));
+                """);
         }
 
         [Fact]
         public async Task Parameter_UsedInConstructorInitializerOnly()
         {
-            await TestDiagnosticMissingAsync(
-                """
+            await VerifyAnalyzerAsync("""
                 class B
                 {
-                    protected B(int p) { }
+                    protected B(int p) { var x = p; }
```

Comment:
I described the reason for the change in the summary of this PR. In this particular case the old test span is around parameter of constructor in the derived class to verify, that the diagnostic isn't reported there. With the new verifier all file is tested now, revealing that this parameter in the base class produces diagnostic as well. I dicided to "disable" diagnostics in such cases with this trick to represent old test behaviour as close as possible and not add additional diagnostic spans, so reader can concentrate on the primary goal of the test without distractions to other "by-product" diagnostics. I can move all this changes to a separate commit, but I don't think this will fulfill the first condition anyway. Therefore I want to ensure that you understood the reason for the change correctly and you are still sure, that such reason isn't enough for making changes in tests. So, should I revert all such changes to tests and replace them with diagnostic markups?

Reply:
Normally for this situation, we would just add the diagnostic spans which were previously missed by the test.

Reply:
Ok, removed all this "suppressions". However, there are still several cases where I fixed compiler errors, which are not connected with the test, e.g. in one test there was `;` missing, which is 100% not the main purpose of that test. I think, such fixes are still valuable.